lib/pushprocessor.js

```diff
@@ -0,0 +1,226 @@
+var micromatch = require("micromatch");
+
+/**
+ * Construct a Push Processor.
+ * @constructor
+ * @param {Object} client The Matrix client object to use
+ */
+module.exports = function(client) {
```

Comment:
This all looks deliciously functional; is there any reason why we need to construct instances of `PushProcessor`?